packages/dd-trace/src/appsec/iast/analyzers/vulnerability-analyzer.js

```diff
@@ -37,10 +58,16 @@ class Analyzer extends Plugin {
 
   analyze (value) {
     const iastContext = getIastContext(storage.getStore())
+    if (!iastContext) return
 
-    if (iastContext && this._isVulnerable(value, iastContext)) {
-      this._report(value, iastContext)
-    }
+    this._reportIfVulnerable(value, iastContext)
+  }
+
+  analyzeAll (...values) {
+    const iastContext = getIastContext(storage.getStore())
+    if (!iastContext) return
+
+    values.forEach(value => this._reportIfVulnerable(value, iastContext))
```

Comment:
If this method has 2 vulnerabilities, we are going to send to the backend only 1 (same file and line), but we will waste time calculating the stack twice.

We should change `_reportIfVulnerable` method to return `true` when it is vulnerable, and break the for:

```suggestion
    for (let i = 0; i < values.length; i++) {
      if (this._reportIfVulnerable(values[i], iastContext)) break;
    }
```

Reply:
Then 'analyzeAll` method will assume that values are from the same file and line... 
Will the condition always be met?

Reply:
Yes, the line and file are calculated in report method, so the stack is always the same.